### PR TITLE
fixed File type custom option value not showing properly on wish list page #24319

### PR DESCRIPTION
**Pull request by rani-priya**
fixed File type custom option value not showing properly on wish list page #24319

### Fixed Issues (if relevant)

1. magento/magento2#24319: fixed File type custom option value not showing properly on wish list page

### Manual testing scenarios (*)

1. create a simple product with file type custom option
2. open product at frontend and then select a file
3. add it to wish list
wishlist-issue

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by VladimirZaets** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, here is your [Magento instance](https://i-24320-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-24320-2-3-develop.instances.magento-community.engineering/admin](https://i-24320-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-5715](https://jira.corp.magento.com/browse/ENGCOM-5715) has been created to process this Pull Request
:eight_spoked_asterisk: @VladimirZaets, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
Hi @rani-webkul 

During testing we faced the issue.

**Problem:** "404 not found" error message occurs in details information when we click on file link

**Steps to reproduce:**

1. Create a simple product with file type custom option;
2. Open product at frontend and then select a file;
3. Add it to wish list;
4. Go to Wish list page;
5. In details information click on file link;

**Actual Result:**
after

@rani-webkul Could you take a look, please?

Thanks!

**Comment by rani-priya** [comment]
@engcom-Alfa  I have made required changes. Please do have a look.

**Comment by rani-priya** [comment]
@VladimirZaets I have made required changes, Please review.

**Comment by rani-priya** [comment]
@VladimirZaets Please review

**Comment by rani-priya** [comment]
@VladimirZaets Please review

File: app/code/Magento/Wishlist/Controller/Index/DownloadCustomOption.php [hunk 1]
```diff
@@ -98,8 +114,8 @@ public function execute()
             if ($secretKey == $info['secret_key']) {
                 $this->_fileResponseFactory->create(
                     $info['title'],
-                    ['value' => $info['quote_path'], 'type' => 'filename'],
-                    DirectoryList::ROOT,
+                    ['value' => $this->_rootDir->getRelativePath($info['quote_path']), 'type' => 'filename'],
+                    $this->rootDirBasePath,
```
Comment (ihor-sviziev) [inline app/code/Magento/Wishlist/Controller/Index/DownloadCustomOption.php]: Hi @rani-webkul,
Wouldn't it be enough to change only `DirectoryList::ROOT` to `DirectoryList::MEDIA`? 

Looks like that was the main issue why it returned no route page and we don't need to change `value`

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
See my comment above

File: app/code/Magento/Wishlist/Controller/Index/DownloadCustomOption.php (hunk 1 above)
Comment (rani-priya) [reply app/code/Magento/Wishlist/Controller/Index/DownloadCustomOption.php]: done..

Comment (ihor-sviziev) [reply app/code/Magento/Wishlist/Controller/Index/DownloadCustomOption.php]: Still it works after this change? :)

Comment (rani-priya) [reply app/code/Magento/Wishlist/Controller/Index/DownloadCustomOption.php]: yes.. it's working.. I have verified it.

Comment (rani-priya) [reply app/code/Magento/Wishlist/Controller/Index/DownloadCustomOption.php]: thanks.. :+1:

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi,

Could you revert all not related changes, for instance adding 
`    protected $rootDirBasePath;`, additional constructor arguments? We don't really need them anymore.
Just to keep only changes 
```php
$option['value'] = $this->escapeHtml($option['value'], ["a"]);
```
and 
```php
DirectoryList::MEDIA,	
```

Also would be great to squash all your changes into single commit, only if you know how to do that.

**Comment by rani-priya** [comment]
@ihor-sviziev  Done in a single commit.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-5715](https://jira.corp.magento.com/browse/ENGCOM-5715) has been created to process this Pull Request

**Comment by rani-priya** [comment]
@ihor-sviziev could you please also review https://github.com/magento/magento2/pull/24390 ?

**Comment by ihor-sviziev** [comment]
Sure, will check it

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**